modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/UserCardTag.java

```diff
@@ -0,0 +1,584 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.frontend.taglib.clay.servlet.taglib.soy.UserCard;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItem;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.LabelItem;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Carlos Lancha
+ */
+public class UserCardTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)request.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		String pathThemeImages = themeDisplay.getPathThemeImages();
+
+		_spritemap = pathThemeImages.concat("/clay/icons.svg");
+
+		return super.doStartTag();
+	}
+
+	public List<DropdownItem> getActionDropdownItems() {
+		if ((_actionDropdownItems == null) && (_userCard != null)) {
+			return _userCard.getActionDropdownItems();
+		}
+
+		return _actionDropdownItems;
+	}
+
+	@Override
+	public String getCssClass() {
+		if ((super.getCssClass() == null) && (_userCard != null)) {
+			return _userCard.getCssClass();
+		}
+
+		return super.getCssClass();
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+	 */
+	@Deprecated
+	@Override
+	public Map<String, String> getData() {
+		if ((super.getData() == null) && (_userCard != null)) {
+			return _userCard.getData();
+		}
+
+		return super.getData();
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+	 */
+	@Deprecated
+	@Override
+	public String getDefaultEventHandler() {
+		if ((super.getDefaultEventHandler() == null) && (_userCard != null)) {
+			return _userCard.getDefaultEventHandler();
+		}
+
+		return super.getDefaultEventHandler();
+	}
+
+	public Boolean getDisabled() {
+		if ((_disabled == null) && (_userCard != null)) {
```

Comment:
Long time ago somebody told me we have this convention: use `== null` for primitives, use `Validator.isNull()` for objects. Is that still correct? If yes, this should be
```
if ((_disabled == null) && Validator.isNotNull(_userCard)) {
```

Reply:
No idea 😅

Reply:
if you use `Validator.isNull` with the wrong type you'll get information when running `gradew formatSource`

Reply:
From the result of `gradlew formatSource`

```
Method 'Validator.isNotNull' should only be used for type 'Long', 'Serializable' or 'String':
```

So I'm leaving this as it is

Reply:
Looking at [Validator](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/Validator.java#L946) implementation, the only difference is when you have `"null"` string
```
"null" == null // false

Validator.isNull("null") // true
```

I honestly think we should not use `Validator` anywhere. It looks to me like it was meant to help with backend stuff like serialization or DAO objects quirks. In our case, using  `Validator` could in fact cause bugs, for example if user sets card title to `"null"`.